modules/ROOT/pages/database/database-connector-examples.adoc

```diff
@@ -0,0 +1,737 @@
+
+DDL statements, or Data Definition Language statements, are special types of SQL statements that do not directly access or modify data.
+Instead, they create, modify, or destroy data structures, stored procedures, and so on. You can also use DDL statements to create or modify data within tables or other data structures.
+
+
+== Create Table Example
+
+Suppose you have a database schema named *Products*. You also have a
+xref:db/db-configure-connection.adoc[connection] named `dbConfig` that is configured to access *Products*.
+The following operation, named `Execute DDL`, creates a new table named *electronic* that has
+columns named *id*, *name*, *description*, *price*, and *discount*:
+
+[source,xml,linenums]
+----
+<db:execute-ddl config-ref="dbConfig">
+    <db:sql>
+        CREATE TABLE electronic(
+            id INT AUTO_INCREMENT PRIMARY KEY,
+            name VARCHAR(100),
+            description VARCHAR(255),
+            price SMALLINT,
+            discount TINYINT
+        )
+    </db:sql>
+</db:execute-ddl>
+----
+
+The `Execute DDL` operation creates a table and returns `0` unless there is an error.
+
+= Example: Execute Bulk Operations - Mule 4
+
+The insert, update, and delete operations can be used for the cases in which each input parameter can take only one value.
+
+For example, when deleting, many rows could match the criteria and get deleted, but only one criteria `(POSITION = X)` is provided.
+The same concept applies for update; if you use `UPDATE PRODUCTS set PRICE = PRICE * 0.9 where PRICE > :price`, you may be applying a
+ 10% discount on many products, but the price input parameter only accepts one value.
+
+If you want to apply different discount rates on products that have different prices, you can do so by executing many operations.
+For example, assume you have a payload which is a list of object of the following structure { price : number, discountRate: number}.
+From that point, you can do the following:
+
+[source,xml,linenums]
+----
+<foreach>
+  <db:update config-ref="dbConfig">
+    <db:input-parameters>
+     #[
+      {
+        'discountRate' : payload.discountRate,
+        'price' : payload.price,
+      }
+    ]
+    </db:input-parameters>
+    <db:sql>
+      UPDATE PRODUCTS set PRICE = PRICE * :discountRate where PRICE > :price
+    </db:sql>
+  </db:update>
+</foreach>
+----
+
+The previous operation accomplishes the task at hand but is inefficient. For each element in the list, one query needs to be executed.
+As such, for each element, the operation must:
+
+* Parse the query
+* Resolve parameters
+* Acquire a connection to the database (either by getting one for the pool or establishing a new one)
+* Pay all the network overhead
+* The RBMS has to process the query and apply changes
+* Release the connection
+
+You can avoid all the above steps by doing a bulk operation. There is only one query in that example; the update statement is constant, not dynamic.
```

Comment:
Hmm, seems like this is good info to have *before* I have read all this. Is there a better spot to discuss bulk operations?

Reply:
Moved a short desc about bulk operations to the top but left the example of bad operations intact.